plugin/trino-phoenix5/src/main/java/io/trino/plugin/phoenix5/PhoenixMergeSink.java

```diff
@@ -13,137 +13,74 @@
-        RowType mergeRowIdColumnType = (RowType) phoenixMergeTableHandle.mergeRowIdColumnHandle().getColumnType();
-        for (RowType.Field field : mergeRowIdColumnType.getFields()) {
-            checkArgument(field.getName().isPresent(), "Merge row id column field must have name");
-            mergeRowIdFieldNamesBuilder.add(getEscapedArgument(field.getName().get()));
-            mergeRowIdFieldTypesBuilder.add(field.getType());
-        }
-        this.mergeRowIdFieldNames = mergeRowIdFieldNamesBuilder.build();
-        this.deleteSink = createDeleteSink(session, mergeRowIdFieldTypesBuilder.build(), phoenixClient, pageSinkId, remoteQueryModifier);
-    }
-
-    private ConnectorPageSink createUpdateSink(
-            ConnectorSession session,
-            PhoenixOutputTableHandle phoenixOutputTableHandle,
-            PhoenixClient phoenixClient,
-            ConnectorPageSinkId pageSinkId,
-            RemoteQueryModifier remoteQueryModifier)
-    {
-        ImmutableList.Builder<String> columnNamesBuilder = ImmutableList.builder();
-        ImmutableList.Builder<Type> columnTypesBuilder = ImmutableList.builder();
-        columnNamesBuilder.addAll(phoenixOutputTableHandle.getColumnNames());
-        columnTypesBuilder.addAll(phoenixOutputTableHandle.getColumnTypes());
-        if (hasRowKey) {
-            columnNamesBuilder.add(ROWKEY);
-            columnTypesBuilder.add(ROWKEY_COLUMN_HANDLE.getColumnType());
-        }
-
-        PhoenixOutputTableHandle updateOutputTableHandle = new PhoenixOutputTableHandle(
-                remoteTableName,
-                columnNamesBuilder.build(),
-                columnTypesBuilder.build(),
-                Optional.empty(),
-                Optional.empty());
-        return new JdbcPageSink(session, updateOutputTableHandle, phoenixClient, pageSinkId, remoteQueryModifier);
-    }
-
-    private ConnectorPageSink createDeleteSink(
-            ConnectorSession session,
-            List<Type> mergeRowIdFieldTypes,
-            PhoenixClient phoenixClient,
-            ConnectorPageSinkId pageSinkId,
-            RemoteQueryModifier remoteQueryModifier)
-    {
-        checkArgument(mergeRowIdFieldNames.size() == mergeRowIdFieldTypes.size(), "Wrong merge row column, columns and types size not match");
-        JdbcOutputTableHandle deleteOutputTableHandle = new PhoenixOutputTableHandle(
-                remoteTableName,
-                mergeRowIdFieldNames,
-                mergeRowIdFieldTypes,
-                Optional.empty(),
-                Optional.empty());
-
-        return new DeleteSink(session, deleteOutputTableHandle, phoenixClient, pageSinkId, remoteQueryModifier);
-    }
-
-    private class DeleteSink
-            extends JdbcPageSink
+    public PhoenixMergeSink(PhoenixClient phoenixClient, RemoteQueryModifier remoteQueryModifier, ConnectorSession session, ConnectorMergeTableHandle connectorMergeTableHandle, ConnectorPageSinkId pageSinkId, QueryBuilder queryBuilder)
```

Comment:
nit. Like above if that declaration would be divided into separate line it would be clear to see that you added one parameter at the end

If you decide to refactor these, please keep it in separate records